pkg/validate/validate.go

```diff
@@ -113,3 +113,15 @@ func URI(uri string) error {
 	}
 	return nil
 }
+
+// URIWithProtocol validates that the URI specifies a certain
+// protocol scheme (e.g. "https")
+func URIWithProtocol(uri string, protocol string) error {
+	parsed, err := url.Parse(uri)
+	if err != nil {
+		return err
+	} else if parsed.Scheme != protocol {
```

Comment:
nit: this else is not required.. 

from https://golang.org/doc/effective_go.html#if
```console
In the Go libraries, you'll find that when an if statement doesn't flow into the next statement—that is, the body ends in break, continue, goto, or return—the unnecessary else is omitted. 
```

Reply:
updated.